dpctl/tensor/libtensor/source/boolean_reductions.hpp

```diff
@@ -244,7 +245,7 @@ py_boolean_reduction(dpctl::tensor::usm_ndarray src,
     all_deps.push_back(copy_metadata_ev);
 
     auto red_ev =
-        fn(exec_q, dst_nelems, red_nelems, src_data, dst_data, dst_nd,
+        fn(exec_q, dst_nelems, red_nelems, src_data, dst_data, iter_nd,
```

Comment:
If `iter_nd` is not the same as `simplified_iter_shape.size()`, perhaps `iter_nd` must be assigned this value. I would think this parameter is what tells the code how many elements of `iter_shape_and_strides` it can access, and changes on lines 238-239 seem to imply it is `3 * simplifier_iter_shape.size()`, not `3 * iter_nd`.

Reply:
I tried applying change 

```diff
diff --git a/dpctl/tensor/libtensor/source/boolean_reductions.hpp b/dpctl/tensor/libtensor/source/boolean_reductions.hpp
index e6d14c8d4..413816f74 100644
--- a/dpctl/tensor/libtensor/source/boolean_reductions.hpp
+++ b/dpctl/tensor/libtensor/source/boolean_reductions.hpp
@@ -235,8 +235,9 @@ py_boolean_reduction(dpctl::tensor::usm_ndarray src,
         std::get<2>(iter_red_metadata_packing_triple_);

     py::ssize_t *iter_shape_and_strides = packed_shapes_and_strides;
+    iter_nd = static_cast<int>(simplified_iter_shape.size());
     py::ssize_t *red_shape_stride =
-        packed_shapes_and_strides + 3 * simplified_iter_shape.size();
+        packed_shapes_and_strides + 3 * iter_nd;

     std::vector<sycl::event> all_deps;
     all_deps.reserve(depends.size() + 1);
``` 

to this PR. Tests are passing too:

```python
import numpy as np, dpctl.tensor as dpt
def check(x_dpt, x_np, axis):
    return np.array_equal(dpt.asnumpy(dpt.all(x_dpt, axis=axis)), np.all(x_np, axis=axis))
def repeat(n):
    res = []
    axes_list = [0, 1, 2, 3, (0,1), (0,2), (0,3), (1,2), (1,3), (0,1,2), (0,1,3), (0,2,3), (1,2,3), (0,1,2,3)]
    for i in range(n):
        x_np = np.random.choice([True, False], size=120).reshape((2,3,4,5))
        x = dpt.asarray(x_np)
        res.append(all([check(x, x_np, a) for a in axes_list]))
    return all(res)

repeat(1000)
# outputs True
```

Reply:
``simplified_iter_shape.size()`` and ``iter_nd`` should be the same, as ``iter_nd`` is what is passed to ``simplify_iteration_space``.

The change to the packing size was only to align structurally with the ``sum`` reduction as much as possible. I can revert it to ``3 * iter_nd``.

Reply:
Yes, you are right. The `simplified_iteration_space` modifies the first argument by reference.  Then please disregard my comment. LGTM!